src/NRedisStack/Search/AggregationRequest.cs

```diff
@@ -123,169 +72,94 @@ public AggregationRequest GroupBy(IList<string> fields, IList<Reducer> reducers)
             return this;
         }
 
-        public AggregationRequest GroupBy(string field, params Reducer[] reducers)
-        {
-            return GroupBy(new string[] { field }, reducers);
-        }
-
         public AggregationRequest GroupBy(Group group)
         {
-            this.groups.Add(group);
+            args.Add(SearchArgs.GROUPBY);
+            group.SerializeRedisArgs(args);
             return this;
         }
 
-        private void GroupBy()
-        {
-            if (groups.Count > 0)
-            {
-                args.Add("GROUPBY");
-                foreach (Group group in groups)
-                {
-                    group.SerializeRedisArgs(args);
-                }
-            }
-        }
-
         public AggregationRequest SortBy(string property) => SortBy(SortedField.Asc(property));
 
-        public AggregationRequest SortBy(params SortedField[] fields)
-        {
-            this.sortedFields.AddRange(fields);
-            return this;
-        }
+        public AggregationRequest SortBy(params SortedField[] fields) => SortBy(-1, fields);
 
-        private void SortBy()
+        public AggregationRequest SortBy(int max, params SortedField[] fields) // TODO: check if it should be params
```

Comment:
I wasn't sure because this signature allows to send only MAX without any field, although at least one field must be sent, thats why I added the `if(fields.Length > 0)`, 
but right now I looked again and this is the syntax of sortby:
`[ SORTBY nargs [ property ASC | DESC [ property ASC | DESC ...]] [MAX num]`
so sending only MAX without any field should be possible.